### PR TITLE
Unless middleware

### PR DESCRIPTION
**Pull request by shzmr**
Unless middleware inspired by [express-unless](https://www.npmjs.com/package/express-unless) middleware.

### `unless(middleware, (UnlessMiddlewareOptions | CustomUnless))`

The `UnlessMiddlewareOptions` object can include:
- `method` - string or array of strings that describe forbidden http methods such as GET, POST, PUT etc...
- `path` - array of strings, Regex and objects that include `url` and `methods` properties, which will be compared against the request.
- `ext` - string or array of strings that describe forbidden path ends (e.g. in `/user/123` it will check against `/123`).

The `CustomUnless` is a function that receives a Request object and returns a boolean. The result of the function will determine if the execution of the middleware is skipped.

### CODE REVIEW

File: packages/unless/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+
+app.listen(3000);
+```
+
+## License
+
+MIT © [v1rtl](https://v1rtl.site)
+
+[npm-badge]: https://img.shields.io/npm/v/@tinyhttp/req?style=flat-square
```
Comment (talentlessguy) [inline packages/unless/README.md:54]: you should change the badge from `req` to `unless`

File: packages/unless/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+//Path example
+app.use(unless(cors(), {path: ['/content/public', /user/, {url: "/public", methods: ['GET']}]});
+
+
+app.listen(3000);
+```
+
+## License
+
+MIT © [v1rtl](https://v1rtl.site)
```
Comment (talentlessguy) [inline packages/unless/README.md:52]: You wrote the module so I guess there should be a license for your name :)

File: packages/unless/package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    },
+    "./package.json": "./package.json",
+    "./": "./"
+  },
+  "keywords": [
+    "tinyhttp",
+    "node.js",
+    "web framework",
+    "web",
+    "middleware"
```
Comment (talentlessguy) [inline packages/unless/package.json:28]: please add the `unless` tag :D

File: packages/unless/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
+import type { Request, Response, NextFunction } from '@tinyhttp/app'
+
+//Options for the middleware
+export interface UnlessMiddlewareOptions {
```
Comment (talentlessguy) [inline packages/unless/src/index.ts:4]: Not sure why it is an interface. It only contains properties, doesn't extend anything and nothing implements it. I'd use `type` instead, wrapped in the `Partial<T>`

**Review by talentlessguy — CHANGES_REQUESTED** [review]

File: packages/unless/src/index.ts [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+//Optional path object
+interface PathObject {
+  url: string
+  methods: string[]
+}
+
+//Middleware interface
+type Middleware = (req: Request, res: Response, next?: NextFunction) => any
```
Comment (talentlessguy) [inline packages/unless/src/index.ts:17]: I forgot to re-export `Middleware` type in `app` package my bad

now it's re-exported so u just need to do `import type { Middleware } from '@tinyhttp/app'`

**Review by talentlessguy — CHANGES_REQUESTED** [review]

**Comment by shzmr** [comment]
Thanks, I will fix it.

**Comment by talentlessguy** [comment]
LGTM, merging! 

will publish a bit later

**Comment by talentlessguy** [comment]
@all-contributors add @shzmr for code

**Pull request merged**